src/main/java/build/buf/protovalidate/RuleCache.java

```diff
@@ -133,37 +119,19 @@ public List<CompiledProgram> compile(
     }
     List<CompiledProgram> programs = new ArrayList<>();
     for (CelRule rule : completeProgramList) {
-      Env ruleEnv = getRuleEnv(fieldDescriptor, message, rule.field, forItems);
-      Variable ruleVar = Variable.newRuleVariable(message, message.getField(rule.field));
-      ProgramOption globals = ProgramOption.globals(ruleVar);
-      Value ruleValue = new ObjectValue(rule.field, message.getField(rule.field));
+      Cel ruleCel = getRuleCel(fieldDescriptor, message, rule.field, forItems);
       try {
-        Program program = ruleEnv.program(rule.astExpression.ast, globals, PARTIAL_EVAL_OPTIONS);
-        Program.EvalResult evalResult = program.eval(Activation.emptyActivation());
-        Val value = evalResult.getVal();
-        if (value != null) {
-          Object val = value.value();
-          if (val instanceof Boolean && value.booleanValue()) {
-            continue;
-          }
-          if (val instanceof String && val.equals("")) {
-            continue;
-          }
-        }
-        Ast residual = ruleEnv.residualAst(rule.astExpression.ast, evalResult.getEvalDetails());
-        programs.add(
-            new CompiledProgram(
-                ruleEnv.program(residual, globals),
-                rule.astExpression.source,
-                rule.rulePath,
-                ruleValue));
```

Comment:
From an internal feature standpoint this is the only step we skipped. 

I am also not sure if it ever worked. It seems we have to tell which variables/attrs are not available by passing a `PartialActivation` type to `eval`. See [here](https://github.com/projectnessie/cel-java/blob/5e68e2486b3c74490367658ca09e1be82fbb37ab/core/src/main/java/org/projectnessie/cel/EvalOption.java#L42).